examples/asset/multi_asset_sync.rs

```diff
@@ -0,0 +1,249 @@
+//! This example illustrates how to wait for multiple assets to be loaded.
+
+use std::{
+    f32::consts::PI,
+    pin::Pin,
+    sync::{
+        atomic::{AtomicU32, Ordering},
+        Arc, Mutex,
+    },
+    task::{Context, Poll, Waker},
+};
+
+use bevy::{gltf::Gltf, prelude::*, tasks::AsyncComputeTaskPool};
+use futures_lite::Future;
+
+/// Holds a bunch of [`Gltf`]s that takes time to load.
+#[derive(Debug, Resource)]
+pub struct OneHundredThings([Handle<Gltf>; 100]);
+
+/// This is required to support both sync and async.
+///
+/// For sync only the easiest implementation is
+/// [`Arc<()>`] and use [`Arc::strong_count`] for completion.
+/// [`Arc<Atomic*>`] is a more robust alternative.
+#[derive(Debug, Resource, Deref)]
+pub struct AssetBarrier(Arc<AssetBarrierInner>);
+
+/// This guard is to be acquired by [`AssetServer::load_acquire`]
+/// and dropped once finished.
+#[derive(Debug, Deref)]
+pub struct AssetBarrierGuard(Arc<AssetBarrierInner>);
+
+/// Tracks how many guards are remaining.
+#[derive(Debug, Resource)]
+pub struct AssetBarrierInner {
+    count: AtomicU32,
+    /// This is not optimal, preferably use `event_listener::Event`
```

Comment:
Explain why this isn't optimal :)

Reply:
It was not lock free :(
Anyway I think polling in `event_listener` as a dev-dependency and do it properly is the correct approach, since it is already a cornerstone of the `smol` crates, hence used in `bevy_task` and `bevy_assets`.